Sources/Datadog/Core/Persistence/FileWriter.swift

```diff
@@ -42,9 +42,9 @@ internal final class FileWriter {
                     try write(data)
                 }
             } else {
+                let atomicData = commaSeparatorData + data
                 try file.append { write in
-                    try write(commaSeparatorData)
-                    try write(data)
+                    try write(atomicData)
                 }
```

Comment:
we can also reduce `try write()` calls
```swift
let rawData = try jsonEncoder.encode(value)
let data: Data = try file.size() == 0 ? (rawData) : (commaSeparator + data)
try file.append { write in try write(data) }
```
wdyt?

Reply:
Good idea 👍. However, to keep the hotfix scope as small as possible, let's do it in a separate PR to `master` (within the "Later improvements" I list in the PR description).